### PR TITLE
Bump OpenTracing version to 0.31.0-RC1

### PR DESCRIPTION
**Pull request by pavolloffay**
@adriancole @devinsba could you please review? 

Note that should be released as a release candidate e.g. `0.23.0-RC1`.

### CODE REVIEW

**Comment by adriancole** [comment]
can take a look tomorrow. Can you handle a nit? Please switch to square code style and/or undo the formatting change (also please scrub out the boiler-plating of inheritDoc) this makes the actual code easier to review.

**Comment by pavolloffay** [comment]
> Please switch to square code style 

Do you mean this? https://github.com/square/java-code-styles. I have installed it and used reformat code from idea. Also remove inheritDoc comments

**Comment by adriancole** [comment]
thx @pavolloffay

File: src/test/java/brave/opentracing/BraveSpanTest.java [hunk 1]
```diff
@@ -57,16 +56,16 @@
...
 
     assertThat(zipkin.spanStore().getTraces())
         .hasSize(1);
   }
 
   @Test public void autoCloseOnTryFinally_doesntReportTwice() {
-    try (ActiveSpan span = tracer.buildSpan("foo").startActive()) {
-      span.deactivate(); // user closes and also auto-close closes
+    try (Scope scope = tracer.buildSpan("foo").startActive()) {
+      scope.close(); // user closes and also auto-close closes
```
Comment (adriancole) [inline src/test/java/brave/opentracing/BraveSpanTest.java]: shouldn't this be scope.span.finish?

File: src/test/java/brave/opentracing/BraveTracerTest.java [hunk 2]
```diff
@@ -13,27 +13,26 @@
...
 import brave.Tracing;
 import brave.propagation.Propagation;
 import brave.propagation.TraceContext;
-import io.opentracing.ActiveSpan;
+import io.opentracing.Scope;
 import io.opentracing.propagation.Format;
 import io.opentracing.propagation.TextMap;
 import io.opentracing.propagation.TextMapExtractAdapter;
 import io.opentracing.propagation.TextMapInjectAdapter;
-import org.junit.Test;
```
Comment (adriancole) [inline src/test/java/brave/opentracing/BraveTracerTest.java]: nit please revert import order

File: src/test/java/brave/opentracing/BraveSpanTest.java [hunk 3]
```diff
@@ -13,30 +13,29 @@
...
-import io.opentracing.ActiveSpan;
+import io.opentracing.Scope;
 import io.opentracing.Span;
 import io.opentracing.SpanContext;
 import io.opentracing.propagation.Format;
 import io.opentracing.propagation.TextMapExtractAdapter;
 import io.opentracing.propagation.TextMapInjectAdapter;
 import io.opentracing.tag.Tags;
 import java.io.IOException;
+import java.util.Collections;
```
Comment (adriancole) [inline src/test/java/brave/opentracing/BraveSpanTest.java]: nit reverse import order

File: src/main/java/brave/opentracing/BraveSpan.java [hunk 4]
```diff
@@ -34,7 +45,111 @@ static BraveSpan wrap(brave.Span span) {
     return new BraveSpan(span);
   }
 
-  private BraveSpan(brave.Span delegate) {
-    super(delegate);
+  /**
+   * Converts an existing {@linkplain Span} for use in OpenTracing apis
+   */
+  public final brave.Span unwrap() {
```
Comment (adriancole) [inline src/main/java/brave/opentracing/BraveSpan.java]: I assume code below here is copy/paste from prior. PS the javadoc here is backwards.

File: src/main/java/brave/opentracing/BraveSpanBuilder.java [hunk 5]
```diff
@@ -81,61 +71,47 @@
...
-  @Override
-  public Span start() {
+  @Override public Span start() {
     return startManual();
   }
 
-  @Override
-  public ActiveSpan startActive() {
+  @Override public Scope startActive() {
+    return startActive(true);
```
Comment (adriancole) [inline src/main/java/brave/opentracing/BraveSpanBuilder.java]: I was wondering about this.. the default behavior is to finish the span when it goes out of scope? Seems opposite unless we expect most instrumentation to wrap synchronous calls that are enclosed by a single code block or method.

Comment (pavolloffay) [reply src/main/java/brave/opentracing/BraveSpanBuilder.java]: Yes, the current scope impl closes it automatically.

I also think that a better sensible default value is *do not finish automatically*. Auto-finish makes sense for end users which rarely write tracing code.

**Comment by pavolloffay** [comment]
I am not sure what is the right import order in this repo. I have put java* into a separate blog just on top of static ones.

**Comment by adriancole** [comment]
>
> I also think that a better sensible default value is *do not finish
> automatically*. Auto-finish makes sense for end users which rarely write
> tracing code.
>
can you see about getting this changed upstream? I'm sure we are not alone
in this opinion and it is important to have sensible defaults. Otherwise
routine instrumentation code gets cluttered with the opposite, especially
due to the race condition on closing the span
https://github.com/Nike-Inc/wingtips/pull/43/files

**Comment by adriancole** [comment]
not technically a race, just it closes in a way that people don't
always see as obvious

**Comment by adriancole** [comment]
>
> I am not sure what is the right import order in this repo. I have put
> java* into a separate blog just on top of static ones.
>
square formatter should correct import order

**Comment by pavolloffay** [comment]
I will raise the issue. So far we have updated only servlet instrumentation and we had to use it there.

**Comment by pavolloffay** [comment]
> square formatter should correct import order

see the last commit https://github.com/openzipkin-contrib/brave-opentracing/pull/55/commits/83307ac686b9c2eb6a22fb5089f0e914a85293d1

**Comment by adriancole** [comment]
@pavolloffay thx there was indeed mixed formats inside. appreciate you sorting that

**Comment by pavolloffay** [comment]
@adriancole do you have any other concerns other than default auto-finish? 

Can I consider this as approved?

File: src/main/java/brave/opentracing/BraveSpanBuilder.java (hunk 5 above)
Comment (adriancole) [reply src/main/java/brave/opentracing/BraveSpanBuilder.java]: what happened to this, by the way? I saw in another issue opentracing people are going to 1.0 soon. Did this policy get reverted or is it locked in (scope stopping the span on close)

**Comment by adriancole** [comment]
I will pull this in, adding a TODO around the auto-finish to revisit it whenever whatever decision happens

**Comment by adriancole** [comment]
going out as 0.23 (since api breaking, minor bump)

**Comment by pavolloffay** [comment]
At this stage I would prefer to not to merge this. 0.31.0 will be out very soon. 

About auto finishing on close: I think we will make default to false.

**Comment by adriancole** [comment]
Haha sorry but it is out! I saw an old nag I missed and thought i was
holding things up having not released it. If a problem i can cut a release
reverting. Lemme know.

**Pull request merged**